capstone/src/main/java/com/google/step/clients/ScoreClient.java

```diff
@@ -0,0 +1,164 @@
+        }
+
+        standardizeScores(scoreMap, restaurantIds);
+        normalizeScores(scoreMap, restaurantIds);
+
+        // Build the scores return list.
+        List<RestaurantScore> scores = new ArrayList<>();
+        for (Long restaurantId: restaurantIds) {
+            scores.add(new RestaurantScore(restaurantId, scoreMap.get(restaurantId)));
+        }
+
+        return Collections.unmodifiableList(scores);
+    }
+
+    /**
+     * Gets the average pageViews for the entire system.
+     * @param allPageViews list of restaurantPageViews holding all the pageViews.
+     * @return double representing the average pageViews for the system.
+     */
+    private double getSystemAverage(List<RestaurantPageViews> allPageViews) {
+        double sum = 0;
+        double count = 0;
+
+        for (RestaurantPageViews restaurant : allPageViews) {
+            List<WeeklyPageView> pageViews = restaurant.getPageViews();
+            count += pageViews.size();
+            for (int i = 0; i < pageViews.size(); i++) {
+                sum += pageViews.get(i).getCount();
+            }
+        }
+
+        return sum / count;
+    }
+
+    /**
+     * Calculate the average page views for a specific restaurant.
+     * @param restaurant has all the pageViews for a specific restaurant.
+     * @return double representing the average pageViews for that restaurant.
+     */
+    private double calculateRestaurantAveragePageViews(RestaurantPageViews restaurant) {
+        List<WeeklyPageView> pageViews = restaurant.getPageViews();
+
+        double sum = 0;
+        double count = pageViews.size();
+
+        for (int i = 0; i < pageViews.size(); i++) {
+            sum += pageViews.get(i).getCount();
+        }
+
+        return sum / count;
+    }
+
+    /**
+     * Calculates the score for a restaurant using the baynesian theorem formula.
+     * @param latestPageViews calculates the score respective to the latest pageViews for the restaurant.
+     * @param restaurantAverage average pageViews for the restaurant.
+     * @param systemAverage average pageViews for the entire system.
+     * @return double representing the score for that specific restaurant.
+     */
+    private double calculateRawScore(int latestPageViews, double restaurantAverage, double systemAverage) {
+        double score = (latestPageViews / (latestPageViews + minimumWeeks)) * restaurantAverage
+                + (minimumWeeks / (latestPageViews + minimumWeeks)) * systemAverage;
+        return score;
+    }
+
+    /**
+     * Standardizes the scores for the entire map of scores using z - score calculations.
+     * @param scoreMap map that holds the restaurant id as a key and the score of that restaurant as a value.
+     * @param restaurantIds list that holds the restaurant ids to get the values of the map.
+     */
+    private void standardizeScores(Map<Long, Double> scoreMap, List<Long> restaurantIds) {
```

Comment:
Is there ever a case that `restaurantIds` has an element that isn't a key of the score map, or vice versa? It might be worth using a single input in this case, and iterating over the entry set of the map instead.

Reply:
Changed so there is no need for a map at all.